app/models/user.rb

```diff
@@ -12,4 +14,22 @@ class User < ActiveRecord::Base
   def display_name
     username.presence || "#{first_name} #{last_name[0]}"
   end
+
+  def name=(name)
+    self.first_name, self.last_name = name.split(' ', 2)
+  end
+
+  def self.from_omniauth(auth_hash)
```

Comment:
Have you considered extracting this public method from here? It doesn't seem to belong here.